### PR TITLE
WIP: [arm64] CoreCLR Implement more SIMD intrinsics

### PR DESCRIPTION
**Pull request by TamarChristinaArm**
This is a WIP for adding more AArch64 SIMD and Scalar intrinsics.

The tbl and tbx assembly are not yet correct as I haven't looked further into
how to handle the register lists.

I am in the process of refactoring the testsuite so that doesn't compile yet either.

This adds the following intrinsics

- ReverseBitOrder
- AbsoluteCompareGreatherThanOrEqual
- AbsoluteCompareGreatherThan
- LeftShiftAndInsert
- RightShiftAndInsert
- ExtractAndNarrowLow
- ExtractAndNarrowHigh
- UnzipEven
- UnzipOdd
- ZipLow
- ZipHigh
- TableVectorLookup
- TableVectorLookupExtension
- TransposeVectorOdd
- TransposeVectorEven
- MultiplyAndSubtract
- MultiplyAndAdd
- AddAcross

### CODE REVIEW

**Comment by BruceForstall** [comment]
cc @echesakovMSFT @CarolEidt @tannergooding

File: src/jit/emitarm64.cpp
Comment (briansull) [inline src/jit/emitarm64.cpp]: The naming convention used for instruction formats require this to be named starting with DV_3 as it encodes three registers

**Review by briansull — CHANGES_REQUESTED** [review]

**Comment by erozenfeld** [comment]
@dotnet/jit-contrib

**Comment by BruceForstall** [comment]
@TamarChristinaArm Can you give us an idea of your plan for getting these finished and checked in?

My expectation is that you will:

1. Wait for https://github.com/dotnet/coreclr/pull/25508 to be merged,
2. Rebase on top of that,
3. Get everything passing,
4. Get it reviewed and merged.

I presume you will get the currently proposed set of intrinsics implemented here merged before starting on additional ones.

Does that match your plan?

**Comment by TamarChristinaArm** [comment]
@BruceForstall 

My plan is

1. Make the required CoreFX changes (early next week)
2. Split off the already accepted Intrinsics to a different patch
3. Submit proposals for the rest
4. Write minimal tests since @tannergooding  is changing the testing framework
5. Push the two PRs

At that point I will leave it up to you guys to decide if you want to wait till #25508 has been accepted or are happy to go ahead.
Either works for me.

I am working on sorting being able to contribute to CoreFX but in the mean time will make the changes so I can test these intrinsics.

**Comment by tannergooding** [comment]
I believe it will be easiest to just wait for #25508 to be merged. The S.P.Corelib changes have already been done (hence the new merge conflicts in this PR) and are now merged on the CoreFX side, so #25508 should become unblocked today.

The remaining work on #25508 is to add support for loading vectors from memory and then writing the basic tests.

**Comment by TamarChristinaArm** [comment]
Ah ok, I'll rebase this one than and get started on getting these in. Thanks!

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in dotnet/coreclr#27549 this repository will be moving to dotnet/runtime on November 13. If you would like to continue working on this PR after this date, the easiest way to move the change to dotnet/runtime is:
1.	In your coreclr repository clone, create patch by running `git format-patch origin`
2.	In your runtime repository clone, apply the patch by running `git apply --directory src/coreclr <path to the patch created in step 1>`

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in #27549 the dotnet/runtime repository will be used going forward for changes to this code base. Closing this PR as no more changes will be accepted into master for this repository. If you’d like to continue working on this change please move it to dotnet/runtime.

**Pull request closed without merging**
